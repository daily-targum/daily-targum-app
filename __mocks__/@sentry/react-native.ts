export const init = () => jest.fn();