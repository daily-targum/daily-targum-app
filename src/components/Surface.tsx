import React from 'react';
import { View, StyleSheet, Platform } from 'react-native';
import { useTheme } from './Theme';
import { BlurView } from 'expo-blur';

const opacityDark = Platform.select({
  ios: 0.6,
  android: 1,
  web: 0.8
});

const opacityLight = Platform.select({
  ios: 0.78,
  android: 1,
  web: 0.8
});

function Surface({
  style,
  innerStyle,
  children,
  backgroundTint,
  tint = 'auto'
}: {
  style?: any,
  innerStyle?: any,
  children?: React.ReactNode,
  backgroundTint?: string,
  tint?: 'dark' | 'light' | 'default' | 'auto'
}) {
  const {dark, colors} = useTheme();
  if(tint == 'auto') {
    tint = dark ? 'light' : 'dark';
  }
  return (
    <BlurView tint={tint} intensity={100} style={style}>
      <View
        style={[StyleSheet.absoluteFillObject, {
          opacity: dark ? opacityDark : opacityLight,
          backgroundColor: backgroundTint || colors.surface
        }, innerStyle]}
      />
      {children}
    </BlurView>
  );
}

export default Surface;