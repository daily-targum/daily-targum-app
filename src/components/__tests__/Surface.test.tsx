import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import Surface from '../Surface';

describe('<Surface />', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <Surface>
        <Text>Hello World</Text>
      </Surface>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
