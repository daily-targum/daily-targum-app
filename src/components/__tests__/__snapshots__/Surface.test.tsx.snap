// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Surface /> renders correctly 1`] = `
<ViewManagerAdapter_ExpoBlurView
  proxiedProperties={
    Object {
      "intensity": 100,
      "tint": "dark",
    }
  }
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "backgroundColor": "#FFFFFF",
          "opacity": 0.78,
        },
        undefined,
      ]
    }
  />
  <Text>
    Hello World
  </Text>
</ViewManagerAdapter_ExpoBlurView>
`;
